Library/Homebrew/cmd/update.sh

```diff
@@ -302,6 +312,9 @@ EOS
   # kill all of subprocess on interrupt
   trap '{ pkill -P $$; wait; exit 130; }' SIGINT
 
+  HOMEBREW_UPDATE_FAILED_FILE="$HOMEBREW_REPOSITORY/.git/UPDATE_FAILED"
```

Comment:
You can safely make it local; it will still be visible to the subshell. Just tested this on my system (both with `/bin/bash` and a more up-to-date Bash from Homebrew).